docs/command-expansion/template-expansion/authoring-templates.mdx

```diff
@@ -0,0 +1,423 @@
+
+**Template**
+
+```
+A{{ subtract-time startTime attributes.arguments.growingDuration }}
+```
+
+which compiles to:
+
+**Result**
+
+```
+A2023-365T23:00:00
+```
+
+#### `format-as-date`
+This helper allows a user to format a date-like string to whatever language is necessary. For example, if I want STOL-like dates (which use `/` instead of a `T`, and include the time zone abbrevriation) but my input is as such:
+**Input**
+
+```
+{
+  ...,
+  startTime: "2025-001T00:00:00.00
+}
+```
+
+then I can write my template as:
+
+**Template**
+
+```
+CMD AT={{ format-as-date startTime }}
+```
+
+which provides me with:
+
+**Result**
+
+```
+CMD AT=2025-001/00:00:00Z
+```
+
+:::info Note
+If the sequencing language is `Text`, it is preferred that users include the Zulu time abbreviation. Currently, if it is left out, and the date is formatted as a YYYY-MM-DD string, the date is automatically converted to the current time zone, which produces some inconsistency in output.
+:::
+
+#### `flatten`
+
+This helper allows a user to format arrays correctly. SeqN array formatting looks a little different than conventional array formatting in Mustache/Handlebars, which we illustrate below:
+**Input**
+
+```
+{
+  ...,
+  attributes: {
+    ...,
+    arguments: {
+      ...,
+      array: [1, 2, 3, 3.5, "string"]
+    }
+  }
+}
+```
+
+**Template**
+
+```
+Unflattened: {{ firstArraySet }}; Flattened: {{ flatten firstArraySet }}
+```
+
+which provides me with:
+
+**Result**
+
+```
+Unflattened: 1,2,3.5,string; Flattened: [1 2 3.5 string]
+```
+
+---
+
+Putting this all together in AERIE, we might have the following:
+
+<figure>
+  <img alt="Aerie UI - Template With Helpers" src={templateHelpers} />
+  <figcaption>Figure 4: A sequence template using helpers</figcaption>
+</figure>
+
+### Other Notes
+
+There are some quirks here, which can be uncovered by looking through the Handlebars documentation. We will list some of the least intuitive ones, however, any quirks that are noticed by users not listed here should be cross-referenced with the real documentation as our listing here cannot be comprehensive.
+
+#### Helpers
+
+One cannot directly write an input that's not in the input object (or in our case, not an existing property of a simulated activity). The following is invalid:
+
+**Input**
+
+```
+<empty>
+```
+
+**Template**
+
+```
+I ate {{ add 1 2 }} bananas.
+```
+
+Similarly, if a non-existent argument (or helper) is used in a template expression, nothing shows up in the result:
+
+**Input**
+
+```
+<empty>
+```
+
+**Template**
+
+```
+I work at {{ company }}.
+```
+
+(which compiles to):
+
+**Result**
+
+```
+I work at .
+```
+
+#### HTML/XML-like String Escaping
+
+HTML escaping (should any inputs coming from a simulated activity have XML-like strings) is automatic in Handlebars. That is, if we have:
+
+**Input**
+```
+{ 
+  name: "<name>"
+}
+```
+
+**Template**
+```
+I am {{ name }}.
+```
+
+it will compile to:
+
+**Result**
+```
+I am &lt;name&gt;.
```

Comment:
Is this **ever** desired behavior for what we're doing? Doesn't seem like it to me. Feels like we should just be recommending triple brackets from the get-go for all of our docs rather than mentioning this as a footnote.